src/compass/utils/elevation_antenna_pattern.py

```diff
@@ -0,0 +1,69 @@
+''' A routine to apply elevation antenna pattern correction
+(EAP correction)'''
+
+
+import numpy as np
+from osgeo import gdal
+
+
+def apply_eap_correction(burst, path_slc_vrt, path_slc_corrected, check_eap):
+    '''
+    Apply Elevation Antenna Pattern correction (EAP correction)
+    on the input burst
+
+    Parameters:
+    -----------
+    burst: Sentinel1BurstSlc
+        Input burst
+    path_slc_vrt: str
+        Path to the burst SLC to be corrected
+    path_slc_corrected: str:
+        Path to the burst SLC after EAP correction
+    check_eap: Namespace
+        A namespace that contains flags if phase and/or magnitute
+        EAP correction are necessary
+
+    Return:
+    -------
+    path_burst_corrected:
+        Path to the GDAL-compatible raster after the correction
+
+    '''
+
+    #TODO Indicate into the metadata to indicate if EAP correction is applied by OEPRA SAS
+
+
+    if not check_eap.phase_correction:
+        print('Antenna pattern correction is not necessary.')
+        return None
```

Comment:
Do we need to return None. I would prefer checking if EAP is necessary in `s1_geocode_slc.py`. If the flag `check_eap` is `True` you execute the `elevation_antenna_pattern.py`. This eliminate the need for an extra argument in the function and is a bit more transparent toward the users

Reply:
Checking the EAP correction take places on this code and `s1_geocode_slc.py`. I agree that that the check here would be redundant. I will remove the lines here.